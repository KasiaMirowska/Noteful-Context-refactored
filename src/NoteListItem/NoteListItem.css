.note {
        background-color: rgb(250, 235, 214); 
        /* background-color:  rgb(37, 37, 88); */
        cursor: pointer; 
        padding: 10px;
        margin-bottom: 15px;
     }
     .li {
        display: flex;
        padding: 10px;
     }
     .note-info-container {
         flex: 1.5;
         height: auto;
         /* text-align: center; */
     }
    .delete-button2 {
        color: black;
        background-color: rgb(250, 235, 214);
         /* background-color:  rgb(37, 37, 88); */
        font-size: 25px;
    }  
    .date {
        display: block;
        padding: 10px;
    }
    
        
    