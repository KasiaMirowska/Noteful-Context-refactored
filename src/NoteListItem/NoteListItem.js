import React from 'react';
import NotefulContext from '../NotefulContext'
import './NoteListItem.css';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';


export default function NoteListItem(props) {
    // console.log(props.modified)
    // const dateStr = props.modified.getFullYear()+'-' + (props.modified.getMonth()+1) + '-'+props.modified.getDate();
   
  
    return (
        <NotefulContext.Consumer>
            {(context) => {
                return (
            <div className='note'>
                <ul >
                    <li id={props.id} >
                        <Link to= {`/openNote/${props.id}`} className="link"> 
                            <h3>{props.name}</h3>
                        </Link>
                        {props.modified}
                        <br />
                        <button 
                            className='delete-button' 
                            type='button'
                            onClick={() => context.deleteNoteRequest(props.id, context.deleteNote)}
                        >
                        Delete Note
                        </button>
                    </li>
                </ul>
                
            </div>
        )}
        }
        </NotefulContext.Consumer>
    )
}

NoteListItem.propTypes = {
    notes: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      folderId: PropTypes.string.isRequired,
      modified: PropTypes.instanceOf(Date),
    }))
  }