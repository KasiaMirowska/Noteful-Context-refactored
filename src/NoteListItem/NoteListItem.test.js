import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NoteListItem from './NoteListItem';
import { BrowserRouter as BR} from 'react-router-dom';

describe('NoteListItem component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BR><NoteListItem id = 'id' name='a name' modified='a date' folderId=' folderId' content='some content'/></BR>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        expect(renderer.create(<BR><NoteListItem id ='id' name='a name' modified='a date' folderId=' folderId' content='some content'/></BR>).toJSON())
        .toMatchSnapshot();
    })

   
})

