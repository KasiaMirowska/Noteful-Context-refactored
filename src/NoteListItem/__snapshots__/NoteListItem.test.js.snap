// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteListItem component renders the UI as expected 1`] = `
<div
  className="note"
>
  <ul>
    <li
      id="id"
    >
      <a
        className="link"
        href="/openNote/id"
        onClick={[Function]}
      >
        <h3>
          a name
        </h3>
      </a>
      NaN-NaN1-NaN
      <br />
      <button
        className="delete-button"
        onClick={[Function]}
        type="button"
      >
        Delete Note
      </button>
    </li>
  </ul>
</div>
`;
